drivers/source/serial_io.c

```diff
@@ -64,16 +72,29 @@ obc_error_code_t sciPrintText(unsigned char *text, uint32_t length) {
 
     /* USER CODE BEGIN */
     // Print text to the serial port using sciSendBytes. Use the mutex to protect the SCI module.
-
+    if (UART_PRINT_REG == scilinREG)
+    {
+        if (sciLinMutex != NULL)
+        {
+            if (xSemaphoreTake(sciLinMutex, UART_MUTEX_BLOCK_TIME) == pdTRUE)
+            {
+                sciSendBytes(text, length);
+                xSemaphoreGive(sciLinMutex);
+            }
+            return OBC_ERR_CODE_SUCCESS;
```

Comment:
Only return success if we sent the data. This should be moved to right after `xSemaphoreGive`